examples/echo.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ *   Wechaty - https://github.com/chatie/wechaty
+ *
+ *   @copyright 2016-2018 Huan LI <zixia@zixia.net>
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+import {
+  EventErrorPayload,
+  EventLoginPayload,
+  EventLogoutPayload,
+  EventMessagePayload,
+  EventScanPayload,
+} from 'wechaty-puppet'
+
+import {PuppetOA} from '../src/mod'
+
+/**
+ *
+ * 1. Declare your Bot!
+ *
+ */
+const puppet = new PuppetOA({
+  port: 80
+});
+
+// 2. Register event handlers for Bot
+puppet
+  .on('logout', onLogout)
+  .on('login', onLogin)
+  .on('scan', onScan)
+  .on('error', onError)
+  .on('message', onMessage)
+
+// 3. Start the bot!
+puppet.start()
+  .catch(async e => {
+    console.error('Bot start() fail:', e)
+    await puppet.stop()
+    process.exit(-1)
+  })
+
+/**
+ *
+ * 4. You are all set. ;-]
+ *
+ */
+
+/**
+ *
+ * 5. Define Event Handler Functions for:
+ *  `scan`, `login`, `logout`, `error`, and `message`
+ *
+ */
+function onScan (payload: EventScanPayload) {
+  if (payload.qrcode) {
+    // Generate a QR Code online via
+    // http://goqr.me/api/doc/create-qr-code/
+    const qrcodeImageUrl = [
+      'https://api.qrserver.com/v1/create-qr-code/?data=',
+      encodeURIComponent(payload.qrcode),
+    ].join('')
+    console.info(`[${payload.status}] ${qrcodeImageUrl}\nScan QR Code above to log in: `)
+  } else {
+    console.info(`[${payload.status}]`)
+  }
+}
+
+function onLogin (payload: EventLoginPayload) {
+  console.info(`${payload.contactId} login`)
+  puppet.messageSendText(payload.contactId, 'Wechaty login').catch(console.error)
```

Comment:
Could you explain what this code is doing?

In my understanding, this is the official account send a message to itself?

Reply:
This code is a copy-pasta from `ding-dong-bot.ts`. I end up with deleting it since it is not really what I want to demonstrate in this example.